### PR TITLE
Stop showing null in NumberEditor

### PR DESCRIPTION
**Pull request by andreadelrio**
And capitalize fields in form.
Covers #463

### CODE REVIEW

File: src/components/PropEditors.js [hunk 1]
```diff
@@ -183,7 +183,7 @@ class NumberEditor extends React.Component {
     return (
       <TextField
         onChange={onChange}
-        value={typeof this.state.value !== "undefined" ? this.state.value : ""}
+        value={typeof this.state.value !== "undefined" && this.state.value !== "null" ? this.state.value : ""}
```
Comment (davidascher) [inline src/components/PropEditors.js]: That tests for string equality with the string `"null"`.  I did that in the case of the other prop only while the backend is sending such strings.  I think the right comparison here is `!== null`

File: src/components/CaseEditor.scss [hunk 2]
```diff
@@ -37,6 +37,7 @@
   margin-top: 14px;
   padding-top:14px;
   display:inline-block;
+  text-transform: capitalize;
```
Comment (davidascher) [inline src/components/CaseEditor.scss]: this is unrelated. I don't mind it, just make sure it's desired ;-)

File: src/components/PropEditors.js (hunk 1 above)
Comment (andreadelrio) [inline src/components/PropEditors.js]: TextField is receiving "null" as a string from `this.state.value`.

File: src/components/CaseEditor.scss (hunk 2 above)
Comment (andreadelrio) [inline src/components/CaseEditor.scss]: Confirming we want this ;)

File: src/components/PropEditors.js (hunk 1 above)
Comment (davidascher) [inline src/components/PropEditors.js]: That's a bug.  Can you work upstream to figure out where that's happening?

File: src/components/PropEditors.js [hunk 3]
```diff
@@ -167,7 +167,7 @@ class NumberEditor extends React.Component {
   }
 
   componentWillReceiveProps(props) {
-    this.setState({ value: String(props.value) });
```
Comment (andreadelrio) [inline src/components/PropEditors.js]: we were converting null to a string here

Comment (davidascher) [inline src/components/PropEditors.js]: I wonder why I did that.  Does the underlying TextField do the stringification if a number is passed (as in, if `props.value` is a number?  (can test by editing a case that has a number in a numeric field)

Comment (andreadelrio) [inline src/components/PropEditors.js]: As far as I can tell, if a number is passed in, it doesn't get converted to a string.

Comment (davidascher) [inline src/components/PropEditors.js]: but the editor still works, right?

Comment (davidascher) [inline src/components/PropEditors.js]: anyway, if with this line


- you can edit a case that has a number field
- you can change the number
- you can save it
- editing it again shows you the new number

then i'm good!

Comment (andreadelrio) [inline src/components/PropEditors.js]: Yes, just tested those bullet points. Looks good.
